aiohttp_swagger3/swagger.py

```diff
@@ -100,37 +90,8 @@ def __init__(
 
         super().__init__()
 
-    def _register_ui(self, ui_settings: Union[SwaggerUiSettings, ReDocUiSettings, RapiDocUiSettings]) -> None:
-        ui_path = ui_settings.path
-        if not ui_path.startswith("/"):
-            raise Exception("path should start with /")
-        need_redirect = ui_path != "/"
-        ui_path = ui_path.rstrip("/")
-        if need_redirect:
-            self._app.router.add_route("GET", ui_path, _redirect)
-
-        if isinstance(ui_settings, SwaggerUiSettings):
-            ui_handler = _swagger_ui
-            ui_template = SWAGGER_UI_TEMPLATE
-            ui_index_html = _SWAGGER_UI_INDEX_HTML
-            dir_name = "swagger_ui"
-        elif isinstance(ui_settings, ReDocUiSettings):
-            ui_handler = _redoc_ui
-            ui_template = REDOC_UI_TEMPLATE
-            ui_index_html = _REDOC_UI_INDEX_HTML
-            dir_name = "redoc_ui"
-        else:
-            ui_handler = _rapidoc_ui
-            ui_template = RAPIDOC_UI_TEMPLATE
-            ui_index_html = _RAPIDOC_UI_INDEX_HTML
-            dir_name = "rapidoc_ui"
-        self._app.router.add_route("GET", f"{ui_path}/", ui_handler)
-        self._app.router.add_route("GET", f"{ui_path}/swagger.json", _swagger_spec)
-
-        base_path = pathlib.Path(__file__).parent
-        self._app.router.add_static(f"{ui_path}/{dir_name}_static", base_path / dir_name)
-
-        self._app[ui_index_html] = ui_template.substitute({"settings": json.dumps(ui_settings.to_settings())})
+    async def _swagger_spec(self, request: web.Request) -> web.Response:
```

Comment:
Why did this need to? I don't see any usage.

Reply:
This is fundamental to making the getting this functionality working. 

Previously the spec handler would access the spec from `request.app[_SWAGGER_SPECIFICATION]`. This only allows for one spec per application.

To allow for multiple specs, the handler needs to access the appropriate spec. I was doing this by using `self.spec`.

I've changed it now so that it accesses it via an argument to the handler, provided by `functools.partial`.   This does reduce the diff size a bit.